### PR TITLE
In the veto plot macro you can now select a single veto group to plot

### PR DESCRIPTION
**Pull request by KonradAltenmueller**
KonradAltenmueller  [](https://gitlab.cern.ch/rest-for-physics/rawlib/-/commits/konrad_commit) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=rest-for-physics&utm_campaign=badge_info)

### CODE REVIEW

**Review by juanangp — APPROVED** [review]

**Review by jgalan — CHANGES_REQUESTED** [review]
If I undersand, the new code intends to return a vector of ids, i.e. `groupIds`. The group ids that correspond to a particular veto.

I would find it better to implement a new method at `TRestRawVetoAnalysisProcess::GetGroupIds(string groupName)`. Implementing all the logic from line 45 to 62 inside.

Then, in your macro could use `veto->GetGroupIds(group);` that returns the group ids. Of course, if group="all" you just return all ids before doing anything else.

If there is no matching group it will return an empty vector.

**Comment by jgalan** [comment]
Also the logic is a bit obscure to me, with those clear vector, etc. You just need to return 

Following https://www.geeksforgeeks.org/how-to-find-index-of-a-given-element-in-a-vector-in-cpp/

you could implement event an additional method:
```
Int_t TRestRawVetoAnalysisProcess::GetGroupIndex(string groupName)
{
    auto it = find(fVetoGroupNames.begin(), fVetoGroupNames.end(), groupName);
 
    if (it != fVetoGroupNames.end())
        return it - v.begin();

    return -1;
}
```

Then, the second method would be as follows in order to match the logic at your macro:

std::vector <string> TRestRawVetoAnalysisProcess::GetGroupIds(string groupName)
{
       std::vector <string> result;
       if (groupName == "all")
                  result = fVetoGroupIds;

       result.push_back(fVetoGroupIds[groupName]);

       return result;
}

However, I would prefer to just return a string, then in your macro, if groupName is "all" just do nothing.

std::vector <string> TRestRawVetoAnalysisProcess::GetGroupIds(string groupName)
{
       return fVetoGroupIds[GetGroupIndex(groupName)];
}

File: macros/REST_Raw_PlotVetoData.C [hunk 1]
```diff
@@ -40,13 +32,25 @@ Int_t REST_Raw_PlotVetoData(string fileName = "data/R01133_000_RawToSignal_Ar2Is
...
+		// check if all groups or only a selected group should be plotted
+		if (group == "all"){
+			pair<vector<string>,vector<string>> vetoGroups = veto->GetVetoGroups();
+			groupNames = std::get<0>(vetoGroups);
+			groupIds   = std::get<1>(vetoGroups);
+		} else {
+			groupNames.push_back(group);
+			groupIds.push_back(veto->GetGroupIds(group));
+			// In case group name doesn't exist:
+			if (std::strcmp(string(groupIds[0]).c_str(),string("-1").c_str())==0){
```
Comment (jgalan) [inline macros/REST_Raw_PlotVetoData.C:47]: I guess it does not need that complex expression.

a `if( groupIds[0] == "-1" )` should be enough. It is just a string comparison.

File: src/TRestRawVetoAnalysisProcess.cxx [hunk 2]
```diff
@@ -385,6 +385,22 @@ TRestEvent* TRestRawVetoAnalysisProcess::ProcessEvent(TRestEvent* evInput) {
...
+		return it - fVetoGroupNames.begin();
+	return -1;
+}
+
+/// \brief Function that returns a string of the signal IDs for the specified veto group
+string TRestRawVetoAnalysisProcess::GetGroupIds(string groupName){
+	Int_t index = GetGroupIndex(groupName);
+	if (index != -1)
+		return fVetoGroupIds[index];
+	return std::string("-1");
```
Comment (jgalan) [inline src/TRestRawVetoAnalysisProcess.cxx:401]: It is a bit rare that a method returning a string returns "-1" when it does not find something. It is more natural that it returns just an empty string "".

```
return "";
```

**Review by jgalan — APPROVED** [review]
This commit could be atomised into different commits.

As an advice for future commits you could have done.

commit 1: "TRestRawVetoAnalysisProcess::GetGroupIndex and GetGroupIds methods added"
commit 2: "REST_Raw_PlotVetoData.C. Addressed javier suggestions at PR#13."
commit 3: "REST_Raw_PlotVetoData.C. Process name argument is now accepted."
commit 4: "REST_Raw_PlotVetoData.C. Cleaning up"
....

Doing that will help tracking changes in the future. No need a long message with explanations.

You may check also the commit message format section at the CONTRIBUTING guide.

https://github.com/rest-for-physics/framework/blob/master/CONTRIBUTING.md

File: macros/REST_Raw_PlotVetoData.C (hunk 1 above)
Comment (KonradAltenmueller) [reply macros/REST_Raw_PlotVetoData.C:47]: I tried this, but it didn't work.

**Pull request merged**